### PR TITLE
Update "Sponsored Images" toggle UI

### PR DESCRIPTION
**Pull request by sangwoo108**
Rough requirements:
- The toggle UI should be below background image options
- The new toggle UI should contain descriptions about Sponsored Images
- When Brave Rewards or Brave Ad are not enabled, a button should be visible
- When a user toggles on the option and Reward & Ads are enabled, the description
  should notify the user that they're earning from it.
- When clicking the "Rewards" button, "Reward tour" UI appears or "Brave Ads" option is enabled.

### Light theme
스크린샷 2022-09-03 오후 8 51 24

### Dark theme
스크린샷 2022-09-03 오후 8 51 47

Resolves https://github.com/brave/brave-browser/issues/25197

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [x] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [x] New files have MPL-2.0 license header
- [x] Adequate test coverage exists to prevent regressions
- [x] Major classes, functions and non-trivial code blocks are well-commented
- [x] Changes in component dependencies are properly reflected in `gn`
- [x] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [x] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
Manual: Described from PR description

### CODE REVIEW

File: components/brave_new_tab_ui/containers/newTab/settings/backgroundImage.tsx [hunk 1]
```diff
@@ -175,6 +167,17 @@ class BackgroundImageSettings extends React.PureComponent<Props, State> {
...
               </StyledCustomBackgroundSettings>
             )}
+            <div style={{ height: '16px' }}/>
+            <SettingsRow>
+              <SponsoredImageToggle
+                onChange={toggleBrandedWallpaperOptIn}
+                onEnableRewards={onEnableRewards}
+                checked={showBackgroundImage && brandedWallpaperOptIn}
+                disabled={!showBackgroundImage /* This option can only be enabled if users opt in for background images */}
+                rewardEnabled={this.props.newTabData.rewardsState.rewardsEnabled}
```
Comment (fallaciousreasoning) [inline components/brave_new_tab_ui/containers/newTab/settings/backgroundImage.tsx:177]: Nit: plural
```suggestion
                rewardsEnabled={this.props.newTabData.rewardsState.rewardsEnabled}
```

File: components/brave_new_tab_ui/containers/newTab/settings/sponsoredImagesToggle.tsx [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+import { SettingsText } from '../../../components/default'
+import { Toggle } from '../../../components/toggle'
+
+// Assets
+import InfoIcon from '../../../components/default/settings/assets/info-icon.svg'
+
+// Utilities
+import { getLocale } from '../../../../common/locale'
+
+const isDarkTheme = (p: any) => {
```
Comment (fallaciousreasoning) [inline components/brave_new_tab_ui/containers/newTab/settings/sponsoredImagesToggle.tsx:19]: Wow the theme type is so awkward :laughing: I had a look to see if we could use it instead but noooope :laughing:

Comment (fallaciousreasoning) [reply components/brave_new_tab_ui/containers/newTab/settings/sponsoredImagesToggle.tsx:19]: optional:
```suggestion
const isDarkTheme = (p: any) => p.theme.name === 'Brave Dark'
```

File: components/brave_new_tab_ui/containers/newTab/settings/sponsoredImagesToggle.tsx [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+  onEnableRewards: () => void
+
+  checked: boolean
+  disabled: boolean
+  rewardEnabled: boolean
+  adsEnabled: boolean
+  canSupportAds: boolean
+}
+
+const Container = styled('div')<{}>`
```
Comment (fallaciousreasoning) [inline components/brave_new_tab_ui/containers/newTab/settings/sponsoredImagesToggle.tsx:34]: Very optional (entirely up to you):
```suggestion
const Container = styled.div`
```

File: components/brave_new_tab_ui/containers/newTab/settings/sponsoredImagesToggle.tsx [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+  background-color: rgba(93, 181, 252, 0.2);
```
Comment (fallaciousreasoning) [inline components/brave_new_tab_ui/containers/newTab/settings/sponsoredImagesToggle.tsx:35]: I take it we aren't lucky enough to have a CSS variable for this color? @petemill is it worth adding the color to `app.global.scss`?

File: components/brave_new_tab_ui/containers/newTab/settings/sponsoredImagesToggle.tsx [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+`
+
+const ToggleRow = styled('div')<{}>`
+  width: 100%;
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+  padding: 12px 14px;
+  border-radius: 10px;
+  background-color: ${p => isDarkTheme(p) ? css`#1E2029` : css`white`};
```
Comment (fallaciousreasoning) [inline components/brave_new_tab_ui/containers/newTab/settings/sponsoredImagesToggle.tsx:51]:
```suggestion
  background-color: var(--background1);
```

should automatically handle light/dark mode (and you can remove `isDarkTheme`)

Comment (fallaciousreasoning) [reply components/brave_new_tab_ui/containers/newTab/settings/sponsoredImagesToggle.tsx:51]: See `brave/components/web-components/app.global.scss` for the CSS variables we have. I think there's a convention for transforming the color name in Figma to the css variable name, but there are a bunch of missing colors in there sadly.

File: components/brave_new_tab_ui/containers/newTab/settings/sponsoredImagesToggle.tsx [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+  grid-column: 2;
+  font-weight: 500;
+  font-size: 11px;
+  line-height: 17px;
+`
+
+const EnableRewardsButton = styled('button')<{}>`
+  width: 236px;
+  height: 40px;
+  background: #4C54D2;
```
Comment (fallaciousreasoning) [inline components/brave_new_tab_ui/containers/newTab/settings/sponsoredImagesToggle.tsx:87]:
```suggestion
  background: var(--interactive5);
```

**Review by fallaciousreasoning — COMMENTED** [review]
Mostly looks good, couple of comments about CSS variables

File: components/brave_new_tab_ui/containers/newTab/settings/sponsoredImagesToggle.tsx (hunk 2 above)
Comment (petemill) [reply components/brave_new_tab_ui/containers/newTab/settings/sponsoredImagesToggle.tsx:19]: Can't use ITheme from brave-ui? But either way we should not detect dark mode this way. We can simply use window.matchMedia if we must use JS for dark mode changes.

Comment (sangwoo108) [reply components/brave_new_tab_ui/containers/newTab/settings/sponsoredImagesToggle.tsx:19]: Ooops, I copied this from https://github.com/brave/brave-core/blob/621d6fdd3b2dba4add21179a542cf6c1b5cddbbb/components/brave_new_tab_ui/components/default/settings/index.ts#L30 without any second thought. If this is not favorable, I guess I can replace this with `@media (prefers-color-scheme)` ?

Comment (sangwoo108) [reply components/brave_new_tab_ui/containers/newTab/settings/sponsoredImagesToggle.tsx:19]: Oh, the only usecase of this function can be replaced with `var(--background1)`. Can be remoed.

File: components/brave_new_tab_ui/containers/newTab/settings/sponsoredImagesToggle.tsx (hunk 4 above)
Comment (sangwoo108) [reply components/brave_new_tab_ui/containers/newTab/settings/sponsoredImagesToggle.tsx:33]: a

File: components/brave_new_tab_ui/containers/newTab/settings/sponsoredImagesToggle.tsx (hunk 3 above)
Comment (sangwoo108) [reply components/brave_new_tab_ui/containers/newTab/settings/sponsoredImagesToggle.tsx]: Cool! done.

File: components/brave_new_tab_ui/containers/newTab/settings/sponsoredImagesToggle.tsx (hunk 5 above)
Comment (sangwoo108) [reply components/brave_new_tab_ui/containers/newTab/settings/sponsoredImagesToggle.tsx]: > convention for transforming the color name in Figma to the css variable name

Great! Is there a doc for this part? I was wondering how we manage all these variable stuffs. Also this is my first time to use Figma, It'd be really nice to learn best practices.

File: components/brave_new_tab_ui/containers/newTab/settings/sponsoredImagesToggle.tsx [hunk 7]
```diff
@@ -0,0 +1,116 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+import * as React from 'react'
+import styled from 'styled-components'
+
+import '$web-components/app.global.scss'
```
Comment (sangwoo108) [inline components/brave_new_tab_ui/containers/newTab/settings/sponsoredImagesToggle.tsx:9]: Q. Is this a right way to use variables?

File: components/brave_new_tab_ui/containers/newTab/settings/sponsoredImagesToggle.tsx [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+  background: url(${InfoIcon}) no-repeat;
+`
+
+const DescriptionTitle = styled.div`
+  font-weight: 600;
+  font-size: 12px;
+  line-height: 18px;
+`
+
+const DescriptionBody = styled.div`
```
Comment (sangwoo108) [inline components/brave_new_tab_ui/containers/newTab/settings/sponsoredImagesToggle.tsx:75]: Q. I wonder if I should prefer more semantic tags over `<div>`. e.g. in this case `<p>` ?

**Comment by diracdeltas** [comment]
probably @PrivacyMatters needs to take a look here to review the strings and overall opt-in UI

**Comment by PrivacyMatters** [comment]
@sangwoo108 It's important to use correct terminology. The images show the toggle set to ON and which makes it an opt-out (not an opt-in). If the toggle was off, and the user turned it ON, it would be opt-in.

The default chosen by Brave is that Sponsored Images are ON by default (so an opt-out), even when Brave Rewards have NOT been enabled by a user.

**Has it been implemented differently in South Korea?**

**Comment by sangwoo108** [comment]
>

**Comment by sangwoo108** [comment]
> @sangwoo108 It's important to use correct terminology. The images show the toggle set to ON and which makes it an opt-out (not an opt-in). If the toggle was off, and the user turned it ON, it would be opt-in.
> 
> The default chosen by Brave is that Sponsored Images are ON by default (so an opt-out), even when Brave Rewards have NOT been enabled by a user.
> 
> **Has it been implemented differently in South Korea?**

Oh, my bad. There's no intended default state of sponsored image.

**Comment by sangwoo108** [comment]
cc. @rebron since strings are from spec.

**Comment by PrivacyMatters** [comment]
@sangwoo108 is this is about providing users who do not have Rewards enabled with a notice that they can earn BAT by enabling it?

File: components/brave_new_tab_ui/containers/newTab/settings/sponsoredImagesToggle.tsx (hunk 4 above)
Comment (fallaciousreasoning) [reply components/brave_new_tab_ui/containers/newTab/settings/sponsoredImagesToggle.tsx:33]: Yeah, so the new design system isn't in Brave yet (though hopefully soon)! For now, I think it's probably fine to either leave this as is, or add a variable to `app.global.scss`. Up to you :smile:

File: components/brave_new_tab_ui/containers/newTab/settings/sponsoredImagesToggle.tsx (hunk 5 above)
Comment (fallaciousreasoning) [reply components/brave_new_tab_ui/containers/newTab/settings/sponsoredImagesToggle.tsx]: I haven't really used it much either :laughing: I think in the new design system the Figma ==> CSS variable transformation is more straightforward. I often just search for the hex color in `app.global.css` or the last bit of the color name in figma but without leading `0s` (ie. `Light Theme/Text/text01` becomes `--text1`).

File: components/brave_new_tab_ui/containers/newTab/settings/sponsoredImagesToggle.tsx (hunk 7 above)
Comment (fallaciousreasoning) [reply components/brave_new_tab_ui/containers/newTab/settings/sponsoredImagesToggle.tsx:9]: Ideally, I think we should add the import in our root container (maybe `brave/components/brave_new_tab_ui/containers/app.tsx`). Then you should be able to use the CSS variables anywhere in the new tab page :smile:

File: components/brave_new_tab_ui/containers/newTab/settings/sponsoredImagesToggle.tsx (hunk 8 above)
Comment (fallaciousreasoning) [reply components/brave_new_tab_ui/containers/newTab/settings/sponsoredImagesToggle.tsx:75]: The problem with `<p>` is that it sometimes has a bit of default styling which might mess up the perfect padding you have (or you might need to add extra styles to override it). I think what you have is good, but it's up to you :smile:

File: components/web-components/app.global.scss [hunk 9]
```diff
@@ -1,5 +1,6 @@
 :root {
   --brave-font-family-default: Poppins, sans-serif;
+  --brave-font-family-non-sherif: Poppins, normal;
```
Comment (fallaciousreasoning) [inline components/web-components/app.global.scss:3]: :+1: thanks for adding this

Comment (fallaciousreasoning) [reply components/web-components/app.global.scss:3]:
```suggestion
  --brave-font-family-non-serif: Poppins, normal;
```

**Comment by sangwoo108** [comment]
> @sangwoo108 is this is about providing users who do not have Rewards enabled with a notice that they can earn BAT by enabling it?

@PrivacyMatters Yes, I think so. This is part of https://github.com/brave/brave-browser/issues/15252 and you can find the spec for this. cc. @rebron

File: components/web-components/app.global.scss (hunk 9 above)
Comment (sangwoo108) [reply components/web-components/app.global.scss:3]: Oh, yeah. This is not a sheriff 😂

File: components/brave_new_tab_ui/containers/newTab/settings/sponsoredImagesToggle.tsx (hunk 8 above)
Comment (sangwoo108) [reply components/brave_new_tab_ui/containers/newTab/settings/sponsoredImagesToggle.tsx:75]: >  you might need to add extra styles to override it

That's exactly why I reverted this to `div`! Then let me leave this as is :)

File: components/brave_new_tab_ui/containers/newTab/settings/sponsoredImagesToggle.tsx (hunk 7 above)
Comment (sangwoo108) [reply components/brave_new_tab_ui/containers/newTab/settings/sponsoredImagesToggle.tsx:9]: Great! Works well :)

File: components/brave_new_tab_ui/containers/newTab/settings/sponsoredImagesToggle.tsx (hunk 5 above)
Comment (sangwoo108) [reply components/brave_new_tab_ui/containers/newTab/settings/sponsoredImagesToggle.tsx]: > Theme/Text/text01 becomes --text1).

Aha! that's the convention! thanks a lot :)

File: components/brave_new_tab_ui/containers/newTab/settings/sponsoredImagesToggle.tsx (hunk 4 above)
Comment (sangwoo108) [reply components/brave_new_tab_ui/containers/newTab/settings/sponsoredImagesToggle.tsx:33]: Cool! Done :)

**Comment by sangwoo108** [comment]
Hi, @PrivacyMatters . Do you have anything else for me to address?

**Comment by PrivacyMatters** [comment]
@sangwoo108 no more comment from me. It is only being presented to people with the SI default ON.

**Comment by sangwoo108** [comment]
Thank you so much @PrivacyMatters ! 

@fallaciousreasoning Could you take another look please?

**Review by fallaciousreasoning — APPROVED** [review]
LGTM, sorry for the delay @sangwoo108

**Comment by sangwoo108** [comment]
> LGTM, sorry for the delay @sangwoo108

No problem at all. You're one of the most responsive reviewers around me so that I count on 😎  Thanks for the review!

**Pull request merged**